src/NumberTheory/vinberg.jl

```diff
@@ -0,0 +1,322 @@
+###################################################################
+# Vinberg's algorithm
+###################################################################
+
+@doc raw"""
+    _check_v0(Q::ZZMatrix, v0::ZZMatrix) -> ZZMatrix
+
+Checks if the inner product of the given `v0` is positive or generates such a `v0` if none is given.
+"""
+function _check_v0(Q::ZZMatrix, v0::ZZMatrix)
+  if iszero(v0)
+    Eigenvalues = eigenvalues(QQ, Q)
+    stopping_condition = 0
+    negative_eigenvalue = 0
+    for x in Eigenvalues
+      if x < 0
+        negative_eigenvalue = x
+        break
+      end
+      stopping_condition =+
+    end
+    @req stopping_condition < length(Eigenvalues) "Q is not of signature (1, n)"
```

Comment:
Actually, if I understand this code right, perhaps you could write it like this?
```suggestion
    ev = eigenvalues(QQ, Q)
    pos = findfirst(is_negative, ev)
    @req pos === nothing "Q is not of signature (1, n)"
    x = ev[pos]
```